### DISCUSSION TITLE
How should I unit test react-native-web code?

### DISCUSSION THREAD

**Discussion by alibalouchi95** [necolas/react-native-web#2341] [Q&A] [answered]
Is there any recommended libraries for unit testing in react native web?
what is the exact configuration for providing test environment?
is there need of using DOM for web tests?

**Comment by yuxizhe** [comment, marked as answer]
Jest     https://jestjs.io/
React Testing Library: https://testing-library.com/docs/react-testing-library/intro/
react-test-renderer  https://reactjs.org/docs/test-renderer.html
react-native-testing-library    https://callstack.github.io/react-native-testing-library/

**Comment by necolas** [comment]
React Native for Web is based on React DOM and the testing setup should mostly be configured as a React DOM project.
